src/FSharp.Data.GraphQL.Client.DesignTime/ProvidedTypesHelper.fs

```diff
@@ -594,9 +592,10 @@ module internal Provider =
             | Some ptype -> ptype
             | None -> failwithf "Expected to find a type \"%s\" on the schema type map, but it was not found." typeName
         schemaTypes
-        |> Seq.map (fun kvp -> kvp.Value)
-        |> Seq.filter (fun itype -> itype.Kind = TypeKind.INTERFACE || itype.Kind = TypeKind.UNION)
-        |> Seq.map (fun itype -> getProvidedType itype.Name, (possibleTypes itype))
+        |> Seq.choose (fun kvp -> 
+            if kvp.Value.Kind = TypeKind.INTERFACE || kvp.Value.Kind = TypeKind.UNION 
+            then Some (getProvidedType kvp.Value.Name, (possibleTypes kvp.Value))
+            else None)
         |> Seq.iter (fun (itype, ptypes) -> ptypes |> Array.iter (fun ptype -> ptype.AddInterfaceImplementation(itype)))
```

Comment:
`Seq.choose` and `Seq.iter` can be fused into a single `Seq.iter`

Reply:
Done.